pkg/controller/che/che_controller.go

```diff
@@ -291,6 +307,252 @@ func (r *ReconcileChe) Reconcile(request reconcile.Request) (reconcile.Result, e
 		return reconcile.Result{}, err
 	}
 
+	// Determine what server groups the API Server knows about
+	groupList, err := r.discoveryClient.ServerGroups()
+	groups := groupList.Groups
+	foundPackagesAPI := false
+	foundOperatorsAPI := false
+	foundKubernetesImagePullerAPI := false
+	for _, group := range groups {
+		if group.Name == packagesv1.SchemeGroupVersion.Group {
+			foundPackagesAPI = true
+		}
+		if group.Name == operatorsv1alpha1.SchemeGroupVersion.Group {
+			foundOperatorsAPI = true
+		}
+		if group.Name == chev1alpha1.SchemeGroupVersion.Group {
+			foundKubernetesImagePullerAPI = true
+		}
+	}
+
+	// If the image puller should be installed but the APIServer doesn't know about PackageManifests/Subscriptions, log a warning and requeue
+	if instance.Spec.ImagePuller.Enable && (!foundPackagesAPI || !foundOperatorsAPI || !foundKubernetesImagePullerAPI) {
+		logrus.Infof("Couldn't find Operator Lifecycle Manager types to install the Kubernetes Image Puller Operator.  Please install Operator Lifecycle Manager to install the operator or disable the image puller by setting spec.imagePuller.enable to false.")
+		return reconcile.Result{Requeue: true}, nil
+	}
+
+	if instance.Spec.ImagePuller.Enable {
```

Comment:
It is possible to move `if` clause to some file ?
`che_controller.go` is already quite big ;)

Reply:
That's what I'm working on right now